src/main/java/com/tysng/duke/domain/Deadline.java

```diff
@@ -0,0 +1,25 @@
+package com.tysng.duke.domain;
+
+import java.util.Date;
+
+/**
+ * This class contains a Date by which the Deadline is due.
+ */
+public class Deadline extends Task {
+    private Date by;
```

Comment:
I understand what you're doing here with the implementation. Perhaps you could rename the variable "by" to "dateToCompleteBy" so that it is more self-explanatory!

Reply:
Wilco, thanks!